source/common/http/conn_pool_base.h

```diff
@@ -88,22 +88,30 @@ class HttpConnPoolImplBase : public Envoy::ConnectionPool::ConnPoolImplBase,
   virtual CodecClientPtr createCodecClient(Upstream::Host::CreateConnectionData& data) PURE;
   Random::RandomGenerator& randomGenerator() { return random_generator_; }
 
-  virtual absl::optional<HttpServerPropertiesCache::Origin> origin() { return {}; }
+  virtual absl::optional<HttpServerPropertiesCache::Origin>& origin() { return origin_; }
   virtual Http::HttpServerPropertiesCacheSharedPtr cache() { return nullptr; }
 
 protected:
   friend class ActiveClient;
 
+  absl::optional<HttpServerPropertiesCache::Origin> origin_;
```

Comment:
nit: This should be private: according to the style guide (which would suggest we need a protected setter.